RoaringBitmap/src/main/java/org/roaringbitmap/BitmapContainer.java

```diff
@@ -1771,20 +1771,8 @@ public void wrap(long[] b) {
 
   @Override
   public void advanceIfNeeded(char minval) {
-    if ((minval) >= (x + 1) * 64) {
-      x = (minval) / 64;
-      w = bitmap[x];
-      while (w == 0) {
-        ++x;
-        if (x == bitmap.length) {
-          return;
-        }
-        w = bitmap[x];
-      }
-    }
-    while (hasNext() && ((peekNext()) < (minval))) {
-      next(); // could be optimized
-    }
+    x = minval >>> 6;
+    w = bitmap[x] & -(1L << minval);
```

Comment:
On second thoughts, this is wrong, `next()` expects zeros to have been skipped over before calling it, but the tests pass, so this suggests a missing test case.